db/db_test.cc

```diff
@@ -4292,6 +4292,67 @@ TEST_F(DBTest, DynamicCompactionOptions) {
 }
 #endif  // ROCKSDB_LITE
 
+// Test dynamic compaction_options_fifo
+TEST_F(DBTest, DynamicFIFOCompactionOptions) {
+  Options options;
```

Comment:
is it difficult to make the test more functional, like verifying at least one of the new options is used for the next compaction? particularly verify `ComputeCompactionScore()` and `FIFOCompactionPicker` have access to the latest options.

also there's a db_options_test.cc where this might fit better, btw.

Reply:
Added functional tests in db_options_test.cc.